### PR TITLE
Remove CHEBI role classes from table

### LINKED ISSUE #87: Fix chemical stimulus trait pattern issues
List of tasks:
- [x] Remove the [CHEBI:50906 role](https://www.ebi.ac.uk/ols/ontologies/chebi/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FCHEBI_50906) classes from filler table [`response_to_chemical_stimulus_trait.tsv`](https://github.com/obophenotype/bio-attribute-ontology/blob/master/src/patterns/data/default/response_to_chemical_stimulus_trait.tsv).
- [x] Replace some of the chemical entities in [`response_to_chemical_stimulus_trait.tsv`](https://github.com/obophenotype/bio-attribute-ontology/blob/master/src/patterns/data/default/response_to_chemical_stimulus_trait.tsv) with chemical entities that are better matched to the GO response terms.
    - CHEBI:30879	alcohol --> CHEBI:16236	ethanol
- [x] Add new response to stimulus by role of chemical template and filler table. Use the existing OBA classes that are removed from the `response to chemical stimulus by chemical entity` template.
- [x] Add variables for `Contributors` to the yaml pattern template.
    - Use a data_list_var + annotation axiom that references it.
    - Use `dcterms:contributor`.
- [x] Check and make sure that all the filler tsv fields (i.e. contributors) work as intended.

See also related issue: <https://github.com/obophenotype/bio-attribute-ontology/issues/53#issuecomment-1111239926_>

### PR DESCRIPTION
**Pull request by rays22**
If applied, this commit will fix #87.

### CODE REVIEW

**Comment by matentzn** [comment]
Wait, we cant merge this - these classes will disappear! They should be stuck into the "chebi role" pattern before this can be merged!

**Comment by rays22** [comment]
> Wait, we cant merge this - these classes will disappear! They should be stuck into the "chebi role" pattern before this can be merged!

Yes, that was my plan too 😺 !

File: src/patterns/data/default/response_to_chemical_stimulus_trait.tsv [hunk 1]
```diff
@@ -1,143 +1,111 @@
...
-OBA:2040134		CHEBI:8707	quetiapine	<https://orcid.org/0000-0002-7073-9172>|<https://orcid.org/0000-0001-8314-2140>
-OBA:2040135		CHEBI:8871	risperidone	<https://orcid.org/0000-0002-7073-9172>|<https://orcid.org/0000-0001-8314-2140>
-OBA:2040136		CHEBI:9150	simvastatin	<https://orcid.org/0000-0002-7073-9172>|<https://orcid.org/0000-0001-8314-2140>
-OBA:2040137		CHEBI:9334	sulfasalazine	<https://orcid.org/0000-0002-7073-9172>|<https://orcid.org/0000-0001-8314-2140>
-OBA:2040138		CHEBI:9398	tamsulosin	<https://orcid.org/0000-0002-7073-9172>|<https://orcid.org/0000-0001-8314-2140>
-OBA:2040139		CHEBI:9448	terbinafine	<https://orcid.org/0000-0002-7073-9172>|<https://orcid.org/0000-0001-8314-2140>
-OBA:2040140		CHEBI:9747	Triptolide	<https://orcid.org/0000-0002-7073-9172>|<https://orcid.org/0000-0001-8314-2140>
-OBA:2040141		CHEBI:9943	venlafaxine	<https://orcid.org/0000-0002-7073-9172>|<https://orcid.org/0000-0001-8314-2140>
+defined_class	defined_class_name	chemical	chemical_name	orcid
+OBA:2040000		CHEBI:10033	warfarin	https://orcid.org/0000-0001-8314-2140
```
Comment (matentzn) [inline src/patterns/data/default/response_to_chemical_stimulus_trait.tsv:2]: What about the other orcids? Seems like you are now only allowing one?

File: src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+---
+pattern_name: response_to_chemical_with_role_stimulus
+
+pattern_iri: http://purl.obolibrary.org/obo/oba/patterns/response_to_chemical_with_role_stimulus.yaml
+
+description: "A trait that affects the biological response to a representative or representatives of a chemical entity group where the members of the group share a role which they can exhibit. Examples of roles include 'drug', 'protease inhibitor', 'opioid agent', etc."
```
Comment (matentzn) [inline src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml:6]: Mouthful... :D

File: src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    vars:
+      - orcid
+
+def:
+  text: "A trait that affects the response to a stimulus with %s."
+  vars:
+    - role
+
+equivalentTo:
+  text: "'process_quality' and 'characteristic_of' some ('response_to_stimulus' and 'has_input' some %s)"
```
Comment (matentzn) [inline src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml:52]: Does this reflect how GO does it?

I would prefer additional redundant () wherever possible, so 

```
('response_to_stimulus' and ('has_input' some %s))
```

File: src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  has_input: RO:0002233
+
+annotationProperties:
+  exact_synonym: oio:hasExactSynonym
+  contributor: dcterms:contributor
+
+vars:
+  role: "'role'"
+
+data_vars:
```
Comment (matentzn) [inline src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml:28]: You didnt figure out to do data_list_vars eh? You should send a message to Huseyin and ask him how to do it

**Review by matentzn — CHANGES_REQUESTED** [review]
Looks great. Please run 

```
sh run.sh make ../patterns/definitions.owl
```

And commit changes to this branch!

File: src/patterns/data/default/response_to_chemical_stimulus_trait.tsv (hunk 1 above)
Comment (rays22) [reply src/patterns/data/default/response_to_chemical_stimulus_trait.tsv:2]: Until the technical issue of adding orcids as data_list_vars is resolved, I can only add one orcid. I will make a ticket and raise this at the tech. team meeting on Monday.

Comment (matentzn) [reply src/patterns/data/default/response_to_chemical_stimulus_trait.tsv:2]: Excellent thank you!

Comment (rays22) [reply src/patterns/data/default/response_to_chemical_stimulus_trait.tsv]: Contributor ORCIDs are fixed now.

File: src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml (hunk 3 above)
Comment (rays22) [reply src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml]: Done.

File: src/patterns/definitions.owl [hunk 5]
```diff
@@ -47942,993 +47944,1277 @@ EquivalentClasses(<http://purl.obolibrary.org/obo/OBA_2001033> ObjectIntersectio
 # Class: <http://purl.obolibrary.org/obo/OBA_2040000> (trait in response to warfarin)
 
 AnnotationAssertion(<http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/OBA_2040000> "A trait that affects the response to a stimulus with warfarin."^^xsd:string)
+AnnotationAssertion(<http://purl.org/dc/terms/contributor> <http://purl.obolibrary.org/obo/OBA_2040000> "https://orcid.org/0000-0001-8314-2140"^^xsd:string)
```
Comment (matentzn) [inline src/patterns/definitions.owl:47947]: orcid should not be a string, it should be an IRI!

File: src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+def:
+  text: "A trait that affects the response to a stimulus with %s."
+  vars:
+    - role
+
+equivalentTo:
+  text: "'process_quality' and
+        ('characteristic_of' some
+        ('response_to_stimulus' and
+        ('has_input' some %s)))"
```
Comment (matentzn) [inline src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml:59]: This does not seem correct.. Shouldnt it be 'has_input' some ('has_role' some %)? Else how is this pattern different than the other one?

**Review by matentzn — CHANGES_REQUESTED** [review]

File: src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+def:
+  text: "A trait that affects the response to a stimulus with %s."
+  vars:
+    - role
+
+equivalentTo:
+  text: "'process_quality' and
+        ('characteristic_of' some
+        ('response_to_stimulus' and
+        ('has_input' some ('has_role' some %))))"
```
Comment (matentzn) [inline src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml:60]: Just to be fully aligned with upheno:

https://raw.githubusercontent.com/obophenotype/upheno/master/src/patterns/dosdp-dev/abnormalLevelOfChemicalEntityWithRoleInLocation.yaml

```
('has_input' some ('chemical entity' and ('has_role' some %s)))))
```
?

File: src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+annotationProperties:
+  exact_synonym: oio:hasExactSynonym
+  contributor: dct:contributor  # contributor: dcterms:contributor
+
+vars:
+  role: "'role'"
+
+data_list_vars:
+  orcid: xsd:string  # ORCID of contributor
```
Comment (matentzn) [inline src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml:36]: remaining this to figure out.. Should transform to `<https://orcid.org/0000-0002-7356-1779>` in RDF, not `"https://orcid.org/0000-0002-7356-1779"^^xsd:string`

**Review by matentzn — CHANGES_REQUESTED** [review]

**Pull request merged**